### PR TITLE
Split AWSResourcePermission into separate classes

### LINKED ISSUE #1572: [FEATURE]: Split AWSResourcePermission into separate classes
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem statement

AWSResourcePermission is currently too big; this results in many classes referring to this and running into issues of circular dependency. 

### Proposed Solution

In order to avoid this, we need to split this class into smaller classes much like how it's done for Azure

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by nkvuong**
## Changes
- Split ExternalLocationsMigration related functions into a new class

### Linked issues

Resolves #1572

### Tests

- [x] manually tested
- [x] verified on staging environment (screenshot attached)

### CODE REVIEW

**Comment by nkvuong** [comment]
need to wait for #1487 to be merged

File: src/databricks/labs/ucx/aws/access.py [hunk 1]
```diff
@@ -35,7 +34,6 @@ def __init__(
         backend: SqlBackend,
```
Comment (nfx) [inline src/databricks/labs/ucx/aws/access.py:34]: also remove this one

File: src/databricks/labs/ucx/aws/access.py [hunk 2]
```diff
@@ -44,7 +42,6 @@ def __init__(
         self._backend = backend
         self._ws = ws
         self._locations = external_locations
-        self._schema = schema
         self._aws_account_id = aws_account_id
         self._kms_key = kms_key
         self._filename = self.INSTANCE_PROFILES_FILE_NAMES
```
Comment (nfx) [inline src/databricks/labs/ucx/aws/access.py:47]: this one is not needed - this is the copy of the constant on the same class

File: src/databricks/labs/ucx/aws/locations.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+class AWSExternalLocationsMigration:
+
+    def __init__(
+        self,
+        ws: WorkspaceClient,
+        external_locations: ExternalLocations,
+        aws_resource_permissions: AWSResourcePermissions,
+    ):
+        self._ws = ws
+        self._locations = external_locations
```
Comment (nfx) [inline src/databricks/labs/ucx/aws/locations.py:24]:
```suggestion
        self._external_locations = external_locations
```

File: src/databricks/labs/ucx/aws/locations.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        external_location_num = 1
+        for path, role_arn in missing_paths:
+            if role_arn not in credential_dict:
+                logger.error(f"Missing credential for role {role_arn} for path {path}")
+                continue
+            while True:
+                external_location_name = f"{location_init}_{external_location_num}"
+                if external_location_name not in external_location_names:
+                    break
+                external_location_num += 1
```
Comment (nfx) [inline src/databricks/labs/ucx/aws/locations.py:47]: can you turn these lines into a private method? there might be a list of infinite loop and it's always easier to find one with stack traces if it's in an isolated method

**Review by nfx — APPROVED** [review]

File: src/databricks/labs/ucx/aws/access.py (hunk 2 above)
Comment (nkvuong) [reply src/databricks/labs/ucx/aws/access.py]: removed

File: src/databricks/labs/ucx/aws/access.py (hunk 1 above)
Comment (nkvuong) [reply src/databricks/labs/ucx/aws/access.py]: removed

**Comment by nfx** [comment]
@nkvuong what can be done to remove flakiness here?

**Review by nfx — APPROVED** [review]

**Pull request merged**